### PR TITLE
Add Calico helm chart

### PR DESCRIPTION
**Pull request by manuelbuil**
This is the first version of the rke2-calico helm chart.

Compared to the upstream default values, this chart has the following default values (can be modified via helm):
- disables bgp
- uses VXLAN for encapsulation
- pods cidr: 10.42.0.0./16

The helm chart fetches the upstream released tarball with version 3.18.1 and separates it into two charts:

- rke2-calico-crd: Contains the 3 required crds for the operator 
- rke2-calico: Contains the operator and the crds required for calico and managed by the operator

Limitations:
- Only tested on Ubuntu VMs using libvirt
- Basic non-defaulted features not tested (e.g. bgp) 
- Advanced features not tested (Ipv6, encryption, eBPF...) not tested
- Upgrades not tested
- cidr not consuming `--cluster-cidr`
Signed-off-by: Manuel Buil <mbuil@suse.com>

### CODE REVIEW

**Comment by manuelbuil** [comment]
Tested on 1 controller / 2 workers. I can do intrahost ping and interhost ping. I can reach a HTTP service and network policies work too:
```
NAMESPACE         NAME                                                 READY   STATUS      RESTARTS   AGE   IP              NODE      NOMINATED NODE   READINESS GATES
calico-system     calico-kube-controllers-77c7dbc6d6-62lj5             1/1     Running     0          47m   10.42.219.70    master    <none>           <none>
calico-system     calico-node-f7wtz                                    1/1     Running     0          42m   10.17.3.3       worker2   <none>           <none>
calico-system     calico-node-j4r6l                                    1/1     Running     0          47m   10.17.3.1       master    <none>           <none>
calico-system     calico-node-j94cw                                    1/1     Running     0          43m   10.17.3.2       worker1   <none>           <none>
calico-system     calico-typha-74478cc44d-4ctdf                        1/1     Running     0          43m   10.17.3.2       worker1   <none>           <none>
calico-system     calico-typha-74478cc44d-6rrhz                        1/1     Running     0          42m   10.17.3.3       worker2   <none>           <none>
calico-system     calico-typha-74478cc44d-rjgxp                        1/1     Running     0          47m   10.17.3.1       master    <none>           <none>
default           httpbin-79dd8457cc-v6jqz                             1/1     Running     0          34m   10.42.235.130   worker1   <none>           <none>
default           multitool-deployment-79d98599fb-6h564                1/1     Running     0          38m   10.42.189.65    worker2   <none>           <none>
default           multitool-deployment-79d98599fb-94ctf                1/1     Running     0          38m   10.42.189.66    worker2   <none>           <none>
default           multitool-deployment-79d98599fb-cslbs                1/1     Running     0          38m   10.42.235.129   worker1   <none>           <none>
kube-system       etcd-master                                          1/1     Running     0          49m   10.17.3.1       master    <none>           <none>
kube-system       helm-install-rke2-calico-52f89                       0/1     Completed   0          47m   10.17.3.1       master    <none>           <none>
kube-system       helm-install-rke2-calico-crds-h2hjj                  0/1     Completed   0          48m   10.17.3.1       master    <none>           <none>
kube-system       helm-install-rke2-coredns-snv49                      0/1     Completed   0          49m   10.17.3.1       master    <none>           <none>
kube-system       helm-install-rke2-ingress-nginx-62vxq                0/1     Completed   0          49m   10.42.219.66    master    <none>           <none>
kube-system       helm-install-rke2-kube-proxy-xw7pw                   0/1     Completed   0          49m   10.17.3.1       master    <none>           <none>
kube-system       helm-install-rke2-metrics-server-2dr9d               0/1     Completed   0          49m   10.42.219.65    master    <none>           <none>
kube-system       kube-apiserver-master                                1/1     Running     0          50m   10.17.3.1       master    <none>           <none>
kube-system       kube-controller-manager-master                       1/1     Running     0          50m   10.17.3.1       master    <none>           <none>
kube-system       kube-proxy-q79ht                                     1/1     Running     0          42m   10.17.3.3       worker2   <none>           <none>
kube-system       kube-proxy-vx6dv                                     1/1     Running     0          49m   10.17.3.1       master    <none>           <none>
kube-system       kube-proxy-w56df                                     1/1     Running     0          43m   10.17.3.2       worker1   <none>           <none>
kube-system       kube-scheduler-master                                1/1     Running     0          50m   10.17.3.1       master    <none>           <none>
kube-system       rke2-coredns-rke2-coredns-9d9bf756f-cqbdr            1/1     Running     0          49m   10.42.219.69    master    <none>           <none>
kube-system       rke2-ingress-nginx-controller-54946dd48f-8ctqt       1/1     Running     0          47m   10.17.3.1       master    <none>           <none>
kube-system       rke2-ingress-nginx-default-backend-5795954f8-hcvk9   1/1     Running     0          47m   10.42.219.68    master    <none>           <none>
kube-system       rke2-metrics-server-559dc47bd8-clpl5                 1/1     Running     0          47m   10.42.219.67    master    <none>           <none>
tigera-operator   tigera-operator-675ccbb69c-8kzxc                     1/1     Running     0          47m   10.17.3.1       master    <none>           <none>
```

**Review by erikwilson — APPROVED** [review]

**Comment by manuelbuil** [comment]
> > * cidr not consuming `--cluster-cidr`
> 
> Did you want to add that now, or are you going to plumb that through in a separate PR?



> > * cidr not consuming `--cluster-cidr`
> 
> Did you want to add that now, or are you going to plumb that through in a separate PR?

I'll prepare a separate PR tomorrow

File: packages/rke2-calico/generated-changes/patch/Chart.yaml.patch [hunk 1]
```diff
@@ -0,0 +1,12 @@
+--- charts-original/Chart.yaml
++++ charts/Chart.yaml
+@@ -1,5 +1,8 @@
++annotations:
++  # Operator is installed in tigera-operator ns but calico components are installed in calico-system
```
Comment (brandond) [inline packages/rke2-calico/generated-changes/patch/Chart.yaml.patch:5]: Is this comment accurate? Looks like they're going into rke2-calico ns.

File: packages/rke2-calico/generated-changes/patch/values.yaml.patch [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+@@ -3,6 +3,12 @@
+ installation:
+   enabled: true
+   kubernetesProvider: ""
++  calicoNetwork:
++    bgp: Disabled
++    ipPools:
++    - natOutgoing: Enabled
++      encapsulation: VXLAN
++      cidr: 10.42.0.0/16
```
Comment (brandond) [inline packages/rke2-calico/generated-changes/patch/values.yaml.patch:12]: Just to be clear, the Calico deployment only supports using its own IPAM, not the Kubernetes IPAM?

File: packages/rke2-calico/generated-changes/patch/Chart.yaml.patch (hunk 1 above)
Comment (manuelbuil) [reply packages/rke2-calico/generated-changes/patch/Chart.yaml.patch:5]: Not sure where you see rke2-calico ns. When looking at the pods:
`calico-system     calico-node-f7wtz                                    1/1     Running     0          42m   10.17.3.3       worker2   <none>           <none>`

Comment (brandond) [reply packages/rke2-calico/generated-changes/patch/Chart.yaml.patch:5]: Apologies, I misread the patch!

**Review by brandond — APPROVED** [review]

File: packages/rke2-calico/generated-changes/patch/values.yaml.patch (hunk 2 above)
Comment (manuelbuil) [reply packages/rke2-calico/generated-changes/patch/values.yaml.patch:12]: There is a way for Calico to support Kubernetes IPAM but it does not work with VXLAN and throws the error: `VXLAN is invalid for ipPool.encapsulation with Calico CNI and HostLocal IPAM` (trying to understand why in parallel). When using VXLAN, we must use Calico own IPAM, which is set by the value you see in values.yaml.patch. I was thinking about using here your`.Values.global.clusterCIDR` to be in sync with the cluster-cidr

**Pull request merged**